blocks/api/test/parser.js

```diff
@@ -97,11 +98,11 @@ describe( 'block parser', () => {
 		it( 'returns false is block is not valid', () => {
 			registerBlockType( 'core/test-block', defaultBlockSettings );
 
-			expect( isValidBlock(
+			expect( () => isValidBlock(
 				'Apples',
 				getBlockType( 'core/test-block' ),
 				{ fruit: 'Bananas' }
-			) ).toBe( false );
+			) ).toThrow();
```

Comment:
To clarify, I think what should be shown in the UI is more like a text-mode diff and a prompt to confirm that the changes are acceptable.